ocaml/idl/datamodel_tracing.ml

```diff
@@ -0,0 +1,239 @@
+        , "endpoints"
+        , "The endpoints that the tracer provider will export spans to. A URL \
+           or the string 'bugtool'. This can refer to an enpoint to the local \
+           file system"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_components =
+  call ~name:"set_components" ~in_oss_since:None ~lifecycle:[]
+    ~doc:
+      "Set the components on which the tracer provider is to registered on. \
+       i.e. xapi, xenopds, networkd, etc"
+    ~params:
+      [
+        (Ref _tracing, "self", "The tracer provider")
+      ; ( Set String
+        , "components"
+        , "The components the tracer provider is to be registered to"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_filters =
+  call ~name:"set_filters" ~in_oss_since:None ~lifecycle:[]
+    ~doc:
+      "Set the filters field of a tracer provider which will dictate what \
+       operations are to be traced"
+    ~params:
+      [
+        (Ref _tracing, "self", "The tracer provider")
+      ; (Set String, "filters", "The filter the tracer provider is to be set to")
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_processors =
+  call ~name:"set_processors" ~in_oss_since:None ~lifecycle:[]
+    ~doc:
+      "Set the processors field of a tracer provider that define what \
+       operations need to take place prior to export"
+    ~params:
+      [
+        (Ref _tracing, "self", "The tracer provider")
+      ; ( Set String
+        , "processors"
+        , "The processors the tracer provider is to be set to"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let t =
+  create_obj ~name:_tracing ~descr:"Describes a tracer provider" ~doccomments:[]
+    ~gen_constructor_destructor:false ~gen_events:true ~in_db:true ~lifecycle:[]
+    ~persist:PersistEverything ~in_oss_since:None
+    ~messages_default_allowed_roles:_R_POOL_ADMIN
+    ~contents:
+      [
+        uid _tracing ~lifecycle:[]
+      ; field ~qualifier:DynamicRO ~ty:(Set (Ref _host)) ~lifecycle:[] "hosts"
+          "A list of hosts the tracer provider is to be registered on"
+          ~ignore_foreign_key:true
+      ; field ~qualifier:DynamicRO ~ty:String ~lifecycle:[] "name_label"
+          "A user-assinged label for a tracer provider"
+      ; field ~qualifier:DynamicRO
+          ~ty:(Map (String, String))
+          ~lifecycle:[] "tags"
+          "Tags to be set on spans to provide extra information on spans to \
+           identify infromation to the end user"
+      ; field ~qualifier:DynamicRO ~ty:(Set String) ~lifecycle:[] "endpoints"
+          "List of endpoints where spans are exported to. Each endpoint is a \
+           URL or the string 'bugtool' refering to our internal logs"
+      ; field ~qualifier:DynamicRO ~ty:(Set String) ~lifecycle:[] "components"
+          "List of componenets the TraverProvider is to be registered on. \
+           Components being programs that have their own memory and different \
```

Comment:
I don't really understand this description. There also seems to be overlap with `filters`.

Reply:
Components refer to daemons that Telemetry should be registered on, filters is a control on what spans are produced.

Reply:
Filters have been removed, Components refer to a xapi side control on the placement o providers, filters/processors are a open telemetry concept for controlling spans that will be exported.